### DISCUSSION TITLE
aws.sqs for ruby :undefined method `resolve_endpoint' for #<Aws::Endpoints::Endpoint:0xxxxxxxxx>

### DISCUSSION THREAD

**Discussion by walkerJia123** [aws/aws-sdk-ruby#2934] [Q&A]
use endpoint_provider , get an error:
undefined method `resolve_endpoint' for #<Aws::Endpoints::Endpoint:0xxxxxxxxx>

this is my code , how can i fix it ?
`  param= Aws::SQS::EndpointParameters.new(
      endpoint: "http://xxx.xxx.xx",
      region: "xxxxx",
      use_dual_stack: false,
      use_fips: false
    )
my_endpoint = Aws::SQS::EndpointProvider.new.resolve_endpoint(param)
 sqs_client = Aws::SQS::Client.new(region: "xxxxxx", credentials: my_credentials, endpoint_provider: my_endpoint)

def message_sent?(sqs_client, queue_url, message_body)
  puts sqs_client
  sqs_client.send_message(
    queue_url: queue_url,
    message_body: message_body,
    message_group_id: "abc",
    message_deduplication_id: "123456"
  )
  true
rescue StandardError => e
  puts "Error sending message: #{e.message}"
  false
end
`
when called method ‘message_sent’, got that error?
error : Error sending message: undefined method `resolve_endpoint' for #<Aws::Endpoints::Endpoint:0xxxxxxxx>

**Comment by mullermp** [comment]
Thanks for opening an issue. Endpoint parameter and providers are useful for resolving dynamic endpoints. It looks like you have a static region and endpoint. Have you tried using the endpoint and region options like this? `Aws::SQS::Client.new(endpoint: '...', region: '...')`

**Reply by walkerJia123** [reply]
yes, i had tried this way, but i got another error: Aws:SQS:Error: NotFound。i dont know what it means?What is it that cannot be found？queue？or others?

**Reply by walkerJia123** [reply]
by the way, I  can successfully call this by Go SDK . This is my first time using Ruby SDK

**Reply by mullermp** [reply]
Can you post a stack trace as well as the code that is causing that error?

**Reply by walkerJia123** [reply]
I‘ll learn how to use  the stack trace in ruby. And I'll show you the results。thank you for your reply。

**Reply by mullermp** [reply]
The only thing I can think of, for SQS not found error, is maybe that your queue name or queue url is incorrect. Are you using a custom :endpoint Client option because you are using a proxy?

**Reply by walkerJia123** [reply]
yes, Im using a proxy.

**Reply by walkerJia123** [reply]
I tried using the endpoint and region options like this:Aws::SQS::Client.new(endpoint: '...', region: '...')；It seems that the endpoint does not take effect, and the path of queueurl is used directly to send the post request. if endpoint='http://1.1.1.1:9001:/test/message' and queueurl='http://1.1.1.1:9001/test/message/myqueuename'. By configuring http_wire_trace: true, the http message shows that the POST address is: test/message/myqueuename, not test/message. why? how can I fix it ?

**Reply by mullermp** [reply]
Yes, queue url is set as the endpoint. I think that is intended. https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-sqs/lib/aws-sdk-sqs/plugins/queue_urls.rb. Does it need to be something different?

One thought I had regarding proxy is that we support proxy endpoints with :http_proxy. You could try using your proxy endpoint with that option and don't configure an :endpoint option.
